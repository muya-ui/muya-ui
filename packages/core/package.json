{
  "name": "@muya-ui/core",
  "version": "0.4.21",
  "author": "Muya UI Team",
  "title": "Muya UI Components",
  "description": "Muya UI - The core library of Muya UI",
  "repository": "https://github.com/muya-ui/muya-ui.git",
  "bugs": {
    "url": "https://github.com/muya-ui/muya-ui/issues"
  },
  "files": [
    "dist",
    "lib",
    "es",
    "typings",
    "types"
  ],
  "main": "lib/index.js",
  "module": "es/index.js",
  "typings": "types/index.d.ts",
  "license": "MIT",
  "scripts": {
    "clean": "cross-env rimraf lib && cross-env rimraf es && cross-env rimraf types && cross-env rimraf tsconfig-build.tsbuildinfo",
    "build": "tsc --version && npm run clean && npm run build:types && npm run build:cjs && npm run build:es",
    "build:types": "cross-env tsc -b ./tsconfig-build.json",
    "build:cjs": "cross-env BABEL_ENV=cjs babel --root-mode upward src -d lib --ignore **/*.test.ts,**/*.test.tsx,**/demos/*.ts,**/demos/*.tsx --extensions \".ts,.tsx\"",
    "build:es": "cross-env BABEL_ENV=es babel --root-mode upward src -d es --ignore **/*.test.ts,**/*.test.tsx,**/demos/*.ts,**/demos/*.tsx --extensions \".ts,.tsx\""
  },
  "peerDependencies": {
    "react": ">=16.8.0",
    "react-dom": ">=16.8.0",
    "@types/react": ">=16.9.0",
    "@types/react-dom": ">=16.9.0",
    "styled-components": ">=4.3.2"
  },
  "peerDependenciesMeta": {
    "@types/react": {
      "optional": true
    },
    "@types/react-dom": {
      "optional": true
    }
  },
  "dependencies": {
    "@muya-ui/utils": "0.1.3",
    "@muya-ui/theme-light": "0.3.18",
    "@types/lodash": "^4.14.149",
    "@types/react-transition-group": "^4.2.0",
    "@types/reactour": "^1.13.1",
    "array-tree-filter": "^2.1.0",
    "async-validator": "^3.2.2",
    "dayjs": "^1.8.15",
    "lodash": "^4.17.15",
    "popper.js": "^1.15.0",
    "react-dropzone": "^10.1.8",
    "react-is": "^16.9.0",
    "react-transition-group": "^4.2.1",
    "reactour": "^1.16.0",
    "resize-observer-polyfill": "^1.5.1",
    "shallow-equal": "^1.2.0",
    "warning": "^4.0.3"
  },
  "devDependencies": {
    "@babel/cli": "^7.5.5",
    "cross-env": "^5.2.0",
    "fetch-jsonp": "^1.1.3",
    "querystring": "^0.2.0",
    "rimraf": "^3.0.0",
    "typescript": "3.5.3"
  },
  "sideEffects": false,
  "engines": {
    "node": ">=8.0.0"
  }
}
