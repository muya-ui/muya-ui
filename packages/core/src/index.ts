/* eslint-disable */
/// <reference path="../typings/index.d.ts" />

// 主题相关
export * from './ThemeProvider';
export * from './utils/useTheme';
export * from './utils/withTheme';
export * from './utils/useStyles';
export * from './utils';

// 交互组件
export * from './Popper';
export * from './Portal';
export * from './Swipe';
export * from './Trigger';

// 动画组件
export * from './Animation';

// 基础组件
export * from './Alert';
export * from './AutoComplete';
export * from './Badge';
export * from './Breadcrumbs';
export * from './Background';
export * from './Button';
export * from './Calendar';
export * from './Carousel';
export * from './Cascader';
export * from './Checkbox';
export * from './DatePicker';
export * from './DateTimePicker';
export * from './WeekPicker';
export * from './Dialog';
export * from './Drawer';
export * from './Dropdown';
export * from './EllipsisContent';
export * from './Form';
export * from './Grid';
export * from './Img';
export * from './ImgCropper';
export * from './ImgPreview';
export * from './Input';
export * from './InputNumber';
export * from './IconButton';
export * from './Link';
export * from './Menu';
export * from './Notification';
export * from './Popconfirm';
export * from './PopoverCard';
export * from './Progress';
export * from './Radio';
export * from './Select';
export * from './Slider';
export * from './Skeleton';
export * from './Spin';
export * from './Steps';
export * from './Switch';
export * from './Pagination';
export * from './Result';
export * from './Table';
export * from './Tabs';
export * from './Tag';
export * from './toast';
export * from './Tooltip';
export * from './TimePicker';
export * from './Tree';
export * from './TreeSelect';
export * from './Typography';
export * from './Upload';
export * from './InputGroup';
export * from './Guide';
export * from './Space';
export * from './Collapse';
export * from './ScrollView';
export * from './Sticky';
export * from './Anchor';

// 业务组件
export * from './Card';

// i18n
export * from './Locale';

// 一些通用类型
export { ILabeledValue, IReactElementWithRef } from './types';
