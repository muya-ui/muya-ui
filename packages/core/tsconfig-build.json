{
  "extends": "../../tsconfig.json",
  "include": ["src/**/*"],
  "exclude": [
    "src/**/*.test.ts",
    "src/**/*.test.tsx",
    "**/demos/*.ts",
    "**/demos/*.tsx",
    "typings"
  ],
  "compilerOptions": {
    "rootDir": "./src",
    "outDir": "./types",
    "composite": true,
    "emitDeclarationOnly": true,
    "skipLibCheck": true
  },
  "references": [
    {
      "path": "../utils/tsconfig-build.json"
    },
    {
      "path": "../theme-light/tsconfig-build.json"
    }
  ]
}
