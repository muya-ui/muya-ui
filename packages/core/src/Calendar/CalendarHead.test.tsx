import dayjs from 'dayjs';
import { mount } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';
import sinon from 'sinon';

import { LocaleProvider } from '@muya-ui/core';

import CalendarHead from './CalendarHead';

test('测试普通渲染', () => {
  const tree = renderer
    .create(
      <>
        <LocaleProvider
          locale={{
            'Calendar.yearFormat': 'YYYY',
          }}
        >
          <CalendarHead viewDate={dayjs('2019')} viewType="month" />
          <CalendarHead viewDate={dayjs('2019')} viewType="year" />
          <CalendarHead viewDate={dayjs('2019')} viewType="decade" />
        </LocaleProvider>
        <LocaleProvider
          locale={{
            'Calendar.headOrder': 'month',
            'Calendar.yearFormat': 'YYYY',
          }}
        >
          <CalendarHead viewDate={dayjs('2019')} viewType="month" />
        </LocaleProvider>
      </>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('onSwitch', () => {
  const onSwitch = sinon.spy();
  const wrapper = mount(
    <CalendarHead viewType="month" viewDate={dayjs('2019')} onSwitch={onSwitch} />,
  );
  wrapper
    .find('SwitchButton')
    .first()
    .simulate('click');
  expect(() => {
    sinon.assert.calledWith(onSwitch, 'decade');
  }).not.toThrow();
});
