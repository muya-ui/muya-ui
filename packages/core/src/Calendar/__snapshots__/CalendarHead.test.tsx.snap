// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试普通渲染 1`] = `
Array [
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c3 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c5 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(204,204,204,1);
  font-weight: 400;
  font-size: 12px;
  cursor: pointer;
}

.c5:hover {
  color: rgba(191,191,191,1);
}

.c5:active {
  color: rgba(179,179,179,1);
}

.c5 + .c4 {
  margin-left: 8px;
}

.c2 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  left: 0;
}

.c11 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  right: 0;
}

.c9 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 12px;
  cursor: pointer;
}

.c9:hover {
  color: rgba(51,139,255,1);
}

.c9:active {
  color: rgba(12,108,235,1);
}

.c9 + .c8 {
  margin-left: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 36px;
  border-bottom: 1px solid rgba(238,238,238,1);
}

.c0 .c12 {
  margin-bottom: 0;
}

.c0 .c8 {
  margin: 0;
  padding: 0 2px;
}

.c0 .c1 .c6,
.c0 .c10 .c6 {
  margin: 0 12px;
}

.c0 .c1 .c6 {
  margin-right: 0;
}

.c0 .c10 .c6 {
  margin-left: 0;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-calendar-pager-prev"
    >
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-prev-year"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M577.19 905.575L317.154 524.19a21.67 21.67 0 010-24.378L577.19 118.425a21.67 21.67 0 00-5.743-30.121l-129.8-84.729a21.67 21.67 0 00-29.797 5.851L77.92 499.811a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.796 5.96l129.476-84.512a21.67 21.67 0 006.285-30.013z"
          />
          <path
            d="M945.574 905.575L685.538 524.19a21.67 21.67 0 010-24.378l260.036-381.386a21.67 21.67 0 00-5.742-30.121L809.814 3.792a21.67 21.67 0 00-29.796 5.851L446.305 499.811a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.795 5.96l130.018-84.512a21.67 21.67 0 006.285-30.013z"
          />
        </svg>
      </button>
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-prev-month"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M501.725 499.81l260.036-381.385a21.67 21.67 0 00-5.742-30.121L626 3.575a21.67 21.67 0 00-29.796 5.851l-333.93 490.385a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.796 5.96l129.476-84.512a21.67 21.67 0 006.284-30.013L501.725 524.514a21.67 21.67 0 010-24.703z"
          />
        </svg>
      </button>
    </div>
    <button
      className="c3 c4 c8 c9 muya-calendar-switch-year"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      2019
    </button>
    <button
      className="c3 c4 c8 c9 muya-calendar-switch-month"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      1月
    </button>
    <div
      className="c10 c11 muya-calendar-pager-next"
    >
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-next-month"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M522.187 499.81L262.15 118.426a21.67 21.67 0 015.959-30.121L398.128 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.795 5.96l-129.477-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
          />
        </svg>
      </button>
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-next-year"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M706.379 499.81L446.343 118.426a21.67 21.67 0 015.634-30.121L581.995 3.792a21.67 21.67 0 0129.796 5.851l334.146 490.168a21.67 21.67 0 010 24.378l-333.93 490.276a21.67 21.67 0 01-29.795 5.96l-130.018-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00.433-24.703zm-368.384 0L77.959 118.426a21.67 21.67 0 015.959-30.121L213.936 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.796 5.96L84.351 935.913a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
          />
        </svg>
      </button>
    </div>
  </div>,
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c3 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c5 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(204,204,204,1);
  font-weight: 400;
  font-size: 12px;
  cursor: pointer;
}

.c5:hover {
  color: rgba(191,191,191,1);
}

.c5:active {
  color: rgba(179,179,179,1);
}

.c5 + .c4 {
  margin-left: 8px;
}

.c2 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  left: 0;
}

.c11 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  right: 0;
}

.c9 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 12px;
  cursor: pointer;
}

.c9:hover {
  color: rgba(51,139,255,1);
}

.c9:active {
  color: rgba(12,108,235,1);
}

.c9 + .c8 {
  margin-left: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 36px;
  border-bottom: 1px solid rgba(238,238,238,1);
}

.c0 .c12 {
  margin-bottom: 0;
}

.c0 .c8 {
  margin: 0;
  padding: 0 2px;
}

.c0 .c1 .c6,
.c0 .c10 .c6 {
  margin: 0 12px;
}

.c0 .c1 .c6 {
  margin-right: 0;
}

.c0 .c10 .c6 {
  margin-left: 0;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-calendar-pager-prev"
    >
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-prev-year"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M577.19 905.575L317.154 524.19a21.67 21.67 0 010-24.378L577.19 118.425a21.67 21.67 0 00-5.743-30.121l-129.8-84.729a21.67 21.67 0 00-29.797 5.851L77.92 499.811a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.796 5.96l129.476-84.512a21.67 21.67 0 006.285-30.013z"
          />
          <path
            d="M945.574 905.575L685.538 524.19a21.67 21.67 0 010-24.378l260.036-381.386a21.67 21.67 0 00-5.742-30.121L809.814 3.792a21.67 21.67 0 00-29.796 5.851L446.305 499.811a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.795 5.96l130.018-84.512a21.67 21.67 0 006.285-30.013z"
          />
        </svg>
      </button>
    </div>
    <button
      className="c3 c4 c8 c9 muya-calendar-switch-year"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      2019
    </button>
    <div
      className="c10 c11 muya-calendar-pager-next"
    >
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-next-year"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M706.379 499.81L446.343 118.426a21.67 21.67 0 015.634-30.121L581.995 3.792a21.67 21.67 0 0129.796 5.851l334.146 490.168a21.67 21.67 0 010 24.378l-333.93 490.276a21.67 21.67 0 01-29.795 5.96l-130.018-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00.433-24.703zm-368.384 0L77.959 118.426a21.67 21.67 0 015.959-30.121L213.936 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.796 5.96L84.351 935.913a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
          />
        </svg>
      </button>
    </div>
  </div>,
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c3 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c5 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(204,204,204,1);
  font-weight: 400;
  font-size: 12px;
  cursor: pointer;
}

.c5:hover {
  color: rgba(191,191,191,1);
}

.c5:active {
  color: rgba(179,179,179,1);
}

.c5 + .c4 {
  margin-left: 8px;
}

.c9 {
  box-sizing: border-box;
  margin: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 12px;
  line-height: 16px;
  margin-bottom: 0;
}

.c9 + .c8 {
  margin-top: 0;
}

.c2 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  left: 0;
}

.c12 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  right: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 36px;
  border-bottom: 1px solid rgba(238,238,238,1);
}

.c0 .c10 {
  margin-bottom: 0;
}

.c0 .c13 {
  margin: 0;
  padding: 0 2px;
}

.c0 .c1 .c6,
.c0 .c11 .c6 {
  margin: 0 12px;
}

.c0 .c1 .c6 {
  margin-right: 0;
}

.c0 .c11 .c6 {
  margin-left: 0;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-calendar-pager-prev"
    >
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-prev-year"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M577.19 905.575L317.154 524.19a21.67 21.67 0 010-24.378L577.19 118.425a21.67 21.67 0 00-5.743-30.121l-129.8-84.729a21.67 21.67 0 00-29.797 5.851L77.92 499.811a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.796 5.96l129.476-84.512a21.67 21.67 0 006.285-30.013z"
          />
          <path
            d="M945.574 905.575L685.538 524.19a21.67 21.67 0 010-24.378l260.036-381.386a21.67 21.67 0 00-5.742-30.121L809.814 3.792a21.67 21.67 0 00-29.796 5.851L446.305 499.811a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.795 5.96l130.018-84.512a21.67 21.67 0 006.285-30.013z"
          />
        </svg>
      </button>
    </div>
    <h6
      className="c8 c9 c10 "
    >
      2010-2020
    </h6>
    <div
      className="c11 c12 muya-calendar-pager-next"
    >
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-next-year"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M706.379 499.81L446.343 118.426a21.67 21.67 0 015.634-30.121L581.995 3.792a21.67 21.67 0 0129.796 5.851l334.146 490.168a21.67 21.67 0 010 24.378l-333.93 490.276a21.67 21.67 0 01-29.795 5.96l-130.018-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00.433-24.703zm-368.384 0L77.959 118.426a21.67 21.67 0 015.959-30.121L213.936 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.796 5.96L84.351 935.913a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
          />
        </svg>
      </button>
    </div>
  </div>,
  .c7 {
  display: inline-block;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  font-style: normal;
  line-height: 0;
  text-align: center;
  text-transform: none;
  vertical-align: -0.125em;
  text-rendering: optimizeLegibility;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  width: 1em;
  height: 1em;
  overflow: hidden;
  fill: currentColor;
}

.c3 {
  outline: 0;
  margin: 0;
  font-family: inherit;
  text-transform: none;
  overflow: visible;
  box-sizing: border-box;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  vertical-align: middle;
}

.c5 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(204,204,204,1);
  font-weight: 400;
  font-size: 12px;
  cursor: pointer;
}

.c5:hover {
  color: rgba(191,191,191,1);
}

.c5:active {
  color: rgba(179,179,179,1);
}

.c5 + .c4 {
  margin-left: 8px;
}

.c2 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  left: 0;
}

.c11 {
  position: absolute;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  right: 0;
}

.c9 {
  background: transparent;
  border: none;
  padding: 0;
  color: rgba(51,51,51,1);
  font-weight: 600;
  font-size: 12px;
  cursor: pointer;
}

.c9:hover {
  color: rgba(51,139,255,1);
}

.c9:active {
  color: rgba(12,108,235,1);
}

.c9 + .c8 {
  margin-left: 8px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  height: 36px;
  border-bottom: 1px solid rgba(238,238,238,1);
}

.c0 .c12 {
  margin-bottom: 0;
}

.c0 .c8 {
  margin: 0;
  padding: 0 2px;
}

.c0 .c1 .c6,
.c0 .c10 .c6 {
  margin: 0 12px;
}

.c0 .c1 .c6 {
  margin-right: 0;
}

.c0 .c10 .c6 {
  margin-left: 0;
}

<div
    className="c0"
  >
    <div
      className="c1 c2 muya-calendar-pager-prev"
    >
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-prev-year"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M577.19 905.575L317.154 524.19a21.67 21.67 0 010-24.378L577.19 118.425a21.67 21.67 0 00-5.743-30.121l-129.8-84.729a21.67 21.67 0 00-29.797 5.851L77.92 499.811a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.796 5.96l129.476-84.512a21.67 21.67 0 006.285-30.013z"
          />
          <path
            d="M945.574 905.575L685.538 524.19a21.67 21.67 0 010-24.378l260.036-381.386a21.67 21.67 0 00-5.742-30.121L809.814 3.792a21.67 21.67 0 00-29.796 5.851L446.305 499.811a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.795 5.96l130.018-84.512a21.67 21.67 0 006.285-30.013z"
          />
        </svg>
      </button>
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-prev-month"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M501.725 499.81l260.036-381.385a21.67 21.67 0 00-5.742-30.121L626 3.575a21.67 21.67 0 00-29.796 5.851l-333.93 490.385a21.67 21.67 0 000 24.378l333.93 490.276a21.67 21.67 0 0029.796 5.96l129.476-84.512a21.67 21.67 0 006.284-30.013L501.725 524.514a21.67 21.67 0 010-24.703z"
          />
        </svg>
      </button>
    </div>
    <button
      className="c3 c4 c8 c9 muya-calendar-switch-month"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      1月
    </button>
    <button
      className="c3 c4 c8 c9 muya-calendar-switch-year"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      2019
    </button>
    <div
      className="c10 c11 muya-calendar-pager-next"
    >
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-next-month"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M522.187 499.81L262.15 118.426a21.67 21.67 0 015.959-30.121L398.128 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.795 5.96l-129.477-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
          />
        </svg>
      </button>
      <button
        className="c3 c4 c5 c6 muya-calendar-pager-next-year"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        <svg
          aria-hidden="false"
          className="c7"
          focusable="false"
          viewBox="0 0 1024 1024"
        >
          <path
            d="M706.379 499.81L446.343 118.426a21.67 21.67 0 015.634-30.121L581.995 3.792a21.67 21.67 0 0129.796 5.851l334.146 490.168a21.67 21.67 0 010 24.378l-333.93 490.276a21.67 21.67 0 01-29.795 5.96l-130.018-84.512a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00.433-24.703zm-368.384 0L77.959 118.426a21.67 21.67 0 015.959-30.121L213.936 3.792a21.67 21.67 0 0129.796 5.851l333.82 490.168a21.67 21.67 0 010 24.378l-333.929 490.276a21.67 21.67 0 01-29.796 5.96L84.351 935.913a21.67 21.67 0 01-6.284-30.013l260.036-381.386a21.67 21.67 0 00-.108-24.703z"
          />
        </svg>
      </button>
    </div>
  </div>,
]
`;
